sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/PublishTelemetrySamples.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.DigitalTwins.Core;
+using Azure.DigitalTwins.Core.Samples;
+using static Azure.DigitalTwins.Core.Samples.SampleLogger;
+using static Azure.DigitalTwins.Core.Samples.UniqueIdHelper;
+
+namespace Azure.DigitalTwins.Samples
+{
+    internal class PublishTelemetrySamples
+    {
+        private DigitalTwinsClient DigitalTwinsClient { get; }
+
+        public PublishTelemetrySamples(DigitalTwinsClient dtClient)
+        {
+            DigitalTwinsClient = dtClient;
+        }
+
+        /// <summary>
+        /// Create a temporary component model, twin model and digital twin instance.
+        /// Publish a telemetry message and a component telemetry message to the digital twin instance.
+        /// </summary>
+        public async Task RunSamplesAsync()
+        {
+            PrintHeader("PUBLISH TELEMETRY MESSAGE SAMPLE");
+
+            // For the purpose of this example we will create temporary models using a random model Ids.
+            // We will also create temporary twin instances to publish the telemetry to.
+
+            string componentModelId = await GetUniqueModelIdAsync(SamplesConstants.TemporaryComponentModelPrefix, DigitalTwinsClient).ConfigureAwait(false);
+            string modelId = await GetUniqueModelIdAsync(SamplesConstants.TemporaryModelPrefix, DigitalTwinsClient).ConfigureAwait(false);
+            string twinId = await GetUniqueTwinIdAsync(SamplesConstants.TemporaryTwinPrefix, DigitalTwinsClient).ConfigureAwait(false);
+
+            string newComponentModelPayload = SamplesConstants.TemporaryComponentModelPayload
+                .Replace(SamplesConstants.ComponentId, componentModelId);
+
+            string newModelPayload = SamplesConstants.TemporaryModelPayload
+                .Replace(SamplesConstants.ModelId, modelId)
+                .Replace(SamplesConstants.ComponentId, componentModelId);
+
+            // Then we create the models.
+            await DigitalTwinsClient.CreateModelsAsync(new[] {
+                newComponentModelPayload,
+                newModelPayload
+            }).ConfigureAwait(false);
+
+            Console.WriteLine($"Successfully created models with Ids: {componentModelId}, {modelId}");
+
+            // Create digital twin with Component payload
+            string twinPayload = SamplesConstants.TemporaryTwinPayload
+                .Replace(SamplesConstants.ModelId, modelId)
+                .Replace(SamplesConstants.ComponentId, componentModelId);
+
+            await DigitalTwinsClient.CreateDigitalTwinAsync(twinId, twinPayload).ConfigureAwait(false);
+            Console.WriteLine($"Created digital twin {twinId}.");
+
+            try
+            {
+                #region Snippet:DigitalTwinSamplePublishTelemetry
+
+                Response publishTelemetryResponse = await DigitalTwinsClient.PublishTelemetryAsync(twinId, "\"telemetry\": {\"Telemetry1\": 5}");
```

Comment:
It has to be:

```json
"telemetry": {
  "Key": "value"
}
```

?

The `"telemetry":` part seems weird.

Reply:
Sorry I am not following. The "\"telemetry\": part you commented on is similar to what you put in the comment.

Reply:
I don't understand the required format for supplying telemetry. I figured it would just be a json object with telemetry.

In your example, it seems to require a property called "telemetry" that has a value of said json object.

Is that required?

Reply:
the telemetry payload does not need the "telemetry" key in it; it should be the telemetry payload json object

Reply:
Cool. Will make the change.